packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx

```diff
@@ -0,0 +1,202 @@
+import { formatTimestamp } from '@lightdash/common';
+import {
+    ActionIcon,
+    Alert,
+    Button,
+    CopyButton,
+    Modal,
+    Select,
+    Stack,
+    TextInput,
+    Title,
+    Tooltip,
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { IconAlertCircle, IconCheck, IconCopy } from '@tabler/icons-react';
+import { FC, useEffect, useState } from 'react';
+import useHealth from '../../../hooks/health/useHealth';
+import { useCreateAccessToken } from '../../../hooks/useAccessToken';
+import MantineIcon from '../../common/MantineIcon';
+
+export const CreateTokenModal: FC<{
+    onBackClick: () => void;
+}> = ({ onBackClick }) => {
+    const health = useHealth();
+
+    const {
+        data,
+        mutate: createAccessToken,
+        isLoading,
+        isSuccess,
+    } = useCreateAccessToken();
+
+    const [expireDate, setExpireDate] = useState<Date | undefined>();
+
+    const form = useForm({
+        initialValues: {
+            description: '',
+            expiresAt: '',
+        },
+    });
+
+    const expireOptions = [
+        {
+            label: 'No expiration',
+            value: '',
+        },
+        {
+            label: '7 days',
+            value: '7',
+        },
+        {
+            label: '30 days',
+            value: '30',
+        },
+        {
+            label: '60 days',
+            value: '60',
+        },
+        {
+            label: '90 days',
+            value: '90',
+        },
+    ];
+
+    useEffect(() => {
+        if (isSuccess) {
+            setExpireDate(data.expiresAt);
+        }
+    }, [data?.expiresAt, isSuccess]);
```

Comment:
can you move this to the mutation's onSuccess callback? you can make `useCreateAccessToken` accept mutation args. there's a useAccessToken above that function which accepts query options. in this case it should accept mutation options

Reply:
No need anymore @IrakliJani , I realised I wasn't getting the data correctly, so now I've removed the state usage and reference `data` directly when displaying the expiration on successful creation of token.